test/tax-lot/tax-lot.repository.mock.ts

```diff
@@ -2,22 +2,56 @@ import { generateMock } from "@anatine/zod-mock";
 import {
   findByBblRepoSchema,
   findByBblSpatialRepoSchema,
+  findZoningDistrictByTaxLotBblRepoSchema,
+  checkTaxLotByBblRepoSchema,
 } from "src/tax-lot/tax-lot.repository.schema";
 
 export class TaxLotRepositoryMock {
-  findByBblMocks = Array.from(Array(10), () =>
-    generateMock(findByBblRepoSchema, { seed: 1 }),
+  checkTaxLotByBblMocks = [
+    generateMock(checkTaxLotByBblRepoSchema, { seed: 1 }),
+  ];
+
+  async checkTaxLotByBbl(bbl: string) {
+    return this.checkTaxLotByBblMocks.find((row) => row.bbl === bbl);
+  }
+
+  findByBblMocks = Array.from(this.checkTaxLotByBblMocks, () =>
+    generateMock(findByBblRepoSchema, {
+      stringMap: {
+        bbl: () => this.checkTaxLotByBblMocks[0].bbl,
+      },
+    }),
   );
```

Comment:
Hard coding the 0 makes this a little fragile to change - we may want to add more elements later. Fortunately, we can stop using `Array.from` and switch to `[array].map`. We needed `Array.from` because we were creating arrays from nothing. But now, we can create a new array from an existing array and get direct access to the data in the existing array.

```suggestion
   findByBblMocks = this.checkTaxLotByBblMocks.map((taxLot) =>
    generateMock(findByBblRepoSchema, { stringMap: { bbl: () => taxLot.bbl } }),
  );
```